### PR TITLE
Add FSx for Lustre AutoImport support

### PR DESCRIPTION
**Pull request by gkao123**
**Is this a bug fix or adding new feature?**
This pull request adds support for AutoImport, a new FSxL feature. For more details, please view: https://docs.aws.amazon.com/fsx/latest/LustreGuide/autoimport-data-repo.html

**What testing is done?** 
End to end testing

### CODE REVIEW

File: examples/kubernetes/dynamic_provisioning_s3/README.md [hunk 1]
```diff
@@ -13,12 +13,14 @@ provisioner: fsx.csi.aws.com
...
   subnetId: subnet-056da83524edbe641
   securityGroupIds: sg-086f61ea73388fb6b
+  autoImportPolicy: NONE
   s3ImportPath: s3://ml-training-data-000
   s3ExportPath: s3://ml-training-data-000/export
   deploymentType: SCRATCH_2
 ```
 * subnetId - the subnet ID that the FSx for Lustre filesystem should be created inside.
 * securityGroupIds - a common separated list of security group IDs that should be attached to the filesystem.
+* autoImportPolicy - the policy FSx will follow that determines how the filesystem is automatically updated with changes made in the linked data repository. For list of acceptable policies, please view the official FSx for Lustre documentation
```
Comment (wongma7) [inline examples/kubernetes/dynamic_provisioning_s3/README.md:23]: I tried looking for the documentation myself but couldn't find it. I found https://docs.aws.amazon.com/fsx/latest/LustreGuide/autoimport-data-repo.html but not the strings that each policy corresponds to. Is it OK to link to documentation here or is it a moving target?

**Comment by wongma7** [comment]
lgtm I just have one nitpick about the doc on what values are valid because I couldn't find it myself ...

/approve

File: tests/e2e/dynamic_provisioning_test.go [hunk 2]
```diff
@@ -158,6 +158,7 @@ var _ = Describe("[fsx-csi-e2e] Dynamic Provisioning with s3 data repository", f
 						Parameters: map[string]string{
 							"subnetId":         subnetId,
 							"securityGroupIds": strings.Join(securityGroupIds, ","),
+							"autoImportPolicy": "NEW_CHANGED"
```
Comment (wongma7) [inline tests/e2e/dynamic_provisioning_test.go:161]: think a comma is missing here.

Also might need to run gofmt, the "verify" test complained about it

File: examples/kubernetes/dynamic_provisioning_s3/README.md (hunk 1 above)
Comment (gkao123) [reply examples/kubernetes/dynamic_provisioning_s3/README.md:23]: I can link to this documentation: https://docs.aws.amazon.com/fsx/latest/APIReference/API_CreateFileSystemLustreConfiguration.html!

Comment (wongma7) [reply examples/kubernetes/dynamic_provisioning_s3/README.md:23]: Yeah that would be perfect thanks

File: go.mod [hunk 3]
```diff
@@ -1,7 +1,7 @@
 module github.com/kubernetes-sigs/aws-fsx-csi-driver
 
 require (
-	github.com/aws/aws-sdk-go v1.29.9
+	github.com/aws/aws-sdk-go v1.35.6
```
Comment (wongma7) [inline go.mod:4]: try also to run 'go mod tidy', i think go.sum should have been updated too

Comment (gkao123) [reply go.mod:4]: Ran go mod tidy and updated the pr with go.sum. I also verified go make still works.

**Comment by gkao123** [comment]
Hi @wongma7, the e2e tests seem to have failed due to the following error:

 ```
failed to created filesystem CreateFileSystem failed: ServiceLimitExceeded: Account 607362164682 can have at most 100800 GiB of total Lustre storage across file systems. Please contact AWS support to request a limit increase for this account.

```

This seems unrelated to my code change. How should we resolve this error?

**Comment by wongma7** [comment]
I'm cleaning up the test account, some old file systems are still in Deleting state, I'll retest when they're deleted

**Comment by wongma7** [comment]
/retest

should be enough room already

**Comment by gkao123** [comment]
> I'm cleaning up the test account, some old file systems are still in Deleting state, I'll retest when they're deleted



> I'm cleaning up the test account, some old file systems are still in Deleting state, I'll retest when they're deleted

Thank you!

**Comment by wongma7** [comment]
One of the test failed, not sure if it's a flake because it does involve importing from S3: "Dynamic Provisioning with s3 data repository should create a volume on demand with s3 as data repository"
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_aws-fsx-csi-driver/158/pull-aws-fsx-csi-driver-e2e/1314354691160674304

/retest

**Comment by gkao123** [comment]
@wongma7 I noticed in https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_aws-fsx-csi-driver/158/pull-aws-fsx-csi-driver-e2e/1314354691160674304 that the resources are not being successfully deleted after the test is completed. Do you know if this could cause the test failure?

File: tests/e2e/dynamic_provisioning_test.go [hunk 4]
```diff
@@ -158,6 +158,7 @@ var _ = Describe("[fsx-csi-e2e] Dynamic Provisioning with s3 data repository", f
 						Parameters: map[string]string{
 							"subnetId":         subnetId,
 							"securityGroupIds": strings.Join(securityGroupIds, ","),
+							"autoImportPolicy": "NONE",
```
Comment (wongma7) [inline tests/e2e/dynamic_provisioning_test.go:161]: is it needed, shouldnt aws default to NONE?

**Comment by wongma7** [comment]
/lgtm
tests shouldn't affect each other, but we do reuse the same fsx volume for all tests since it would take too long to create one for each test case
I guess the failure was a flake.

File: tests/e2e/dynamic_provisioning_test.go (hunk 4 above)
Comment (gkao123) [reply tests/e2e/dynamic_provisioning_test.go:161]: That's correct that it's not needed and I verified the tests will still pass if it's not included. I included it just to be explicit here but I can remove the parameter if you think it doesn't add value.

Comment (wongma7) [reply tests/e2e/dynamic_provisioning_test.go:161]: thanks no worries, just checking if it was related to test passing :+1:

**Pull request merged**